### PR TITLE
Feature: Add capability for realsense to use depth stream as well

### PR DESCRIPTION
**Pull request by rohitmenon86**
Use aligned depth frames along with rgb frames from realsense for gaussian splatting

### CODE REVIEW

**Comment by muskie82** [comment]
Thanks for your PR! We will test this locally first.

File: configs/live/realsense.yaml
Comment (muskie82) [inline configs/live/realsense.yaml:1]: Change the config file name to "realsense_rgbd.yaml"

File: configs/live/realsense.yaml [hunk 1]
```diff
@@ -1,14 +1,15 @@
 Results:
-  save_results: False
+  save_results: True
   save_dir: "results"
   save_trj: False
   save_trj_kf_intv: 0
   use_gui: True
-  eval_rendering: False
+  eval_rendering: True
```
Comment (muskie82) [inline configs/live/realsense.yaml:7]: Set `save_results` and `eval_rendering` **False** by default.

File: utils/dataset.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import csv
 import glob
 import os
+import time
```
Comment (muskie82) [inline utils/dataset.py:4]: Please remove the time measurement and print out function as it is redundant.

File: utils/dataset.py [hunk 3]
```diff
@@ -428,10 +429,13 @@ class RealsenseDataset(BaseDataset):
     def __init__(self, args, path, config):
         super().__init__(args, path, config)
         self.pipeline = rs.pipeline()
-        self.h, self.w = 720, 1280
+        self.h, self.w = 360, 640
```
Comment (muskie82) [inline utils/dataset.py:432]: Set 720, 1280 as default value

File: utils/dataset.py [hunk 4]
```diff
@@ -444,6 +448,16 @@ def __init__(self, args, path, config):
...
 
+        self.depth_sensor = self.profile.get_device().first_depth_sensor()
+        self.depth_scale  = self.depth_sensor.get_depth_scale()
+        self.depth_profile = rs.video_stream_profile(
+            self.profile.get_stream(rs.stream.depth)
+        )
+        self.depth_intrinsics = self.depth_profile.get_intrinsics()
+        print("Depth Scale is: " , self.depth_scale)
+        print("Depth intrinsics: ", self.depth_intrinsics)
+        print("RGB intrinsics: ", self.rgb_intrinsics)
```
Comment (muskie82) [inline utils/dataset.py:459]: Remove these print functions

File: utils/dataset.py [hunk 5]
```diff
@@ -463,14 +477,27 @@ def __init__(self, args, path, config):
         )
 
         # depth parameters
-        self.has_depth = False
-        self.depth_scale = None
+        self.has_depth = True
```
Comment (muskie82) [inline utils/dataset.py:480]: This hardcoded flag assumes the depth input is always available, but we want to adjust it depending on the sensor_type in config file (either monocular or depth).

File: utils/dataset.py [hunk 6]
```diff
@@ -463,14 +477,27 @@ def __init__(self, args, path, config):
...
+        self.num_frames = 0
+        self.accumulated_time = 0.0
+        self.previous_frame_time = time.time()
+        #self.depth_scale = None
```
Comment (muskie82) [inline utils/dataset.py:484]: Remove these line

File: utils/dataset.py [hunk 7]
```diff
@@ -463,14 +477,27 @@ def __init__(self, args, path, config):
...
-        rgb_frame = frameset.get_color_frame()
+        aligned_frames = self.align.process(frameset)
+        
+        current_frame_time = time.time()
+        self.num_frames = self.num_frames + 1
+        if self.num_frames > 1:
+            time_diff = (current_frame_time - self.previous_frame_time)
+            self.accumulated_time = self.accumulated_time + time_diff
+            if self.num_frames%50 == 0:
+                print("Inst and Avg FPS", 1/time_diff, self.num_frames/self.accumulated_time)
```
Comment (muskie82) [inline utils/dataset.py:498]: Remove these lines as the timing measurement is redundant.

**Review by muskie82 — CHANGES_REQUESTED** [review]
Thank you for your PR. RGB-D mode works locally, but several changes needed before merging.

- Remove variables/functions for print/logging which are not necessary for the final implementation 
- Modify `dataset.py` to support both monocular and rgb-d inputs depending on the sensor type specified in the config file.

**Comment by rohitmenon86** [comment]
I have made the requested changes. It now works for both monocular mode and depth mode using realsense live stream. Also I tested it with 1280x720 resolution on a different system. Please check if this works for you locally, and if you need any more changes

File: configs/live/realsense.yaml
Comment (rohitmenon86) [reply configs/live/realsense.yaml:1]: Done

File: configs/live/realsense.yaml (hunk 1 above)
Comment (rohitmenon86) [reply configs/live/realsense.yaml]: Done

File: utils/dataset.py (hunk 2 above)
Comment (rohitmenon86) [reply utils/dataset.py]: Done

File: utils/dataset.py (hunk 3 above)
Comment (rohitmenon86) [reply utils/dataset.py]: Done

File: utils/dataset.py (hunk 5 above)
Comment (rohitmenon86) [reply utils/dataset.py]: This comes from the yaml file now

File: utils/dataset.py (hunk 4 above)
Comment (rohitmenon86) [reply utils/dataset.py]: Removed

File: utils/dataset.py (hunk 7 above)
Comment (rohitmenon86) [reply utils/dataset.py]: Removed

**Pull request merged**